app/src/main/java/cz/martykan/forecastie/weatherapi/owm/OpenWeatherMapJsonParser.java

```diff
@@ -0,0 +1,171 @@
+        weather.setPressure(main.getInt("pressure"));
+        weather.setHumidity(main.getInt("humidity"));
+
+        JSONObject weatherJson = weatherObject.getJSONArray("weather").getJSONObject(0);
+
+        String capitalizedDescription = capitalize(weatherJson.getString("description"));
+        weather.setDescription(capitalizedDescription);
+        weather.setWeatherId(weatherJson.getInt("id"));
+
+        setWind(weather, weatherObject.optJSONObject("wind"));
+        setRain(weather, weatherObject.optJSONObject("rain"), weatherObject.optJSONObject("snow"));
+
+        weather.setLastUpdated(Calendar.getInstance().getTimeInMillis());
+
+        return weather;
+    }
+
+    private static void setWind(Weather weather, @Nullable JSONObject windObject) throws JSONException {
+        if (windObject == null) {
+            weather.setWind(0);
+            weather.setWindDirectionDegree(null);
+            return;
+        }
+
+        weather.setWind(windObject.getDouble("speed"));
+
+        if (windObject.has("deg")) {
+            weather.setWindDirectionDegree(windObject.getDouble("deg"));
+        } else {
+            Log.e("parseTodayJson", "No wind direction available");
+            weather.setWindDirectionDegree(null);
+        }
+    }
+
+    private static void setRain(Weather weather, @Nullable JSONObject rainObject, @Nullable JSONObject snowObject) throws JSONException {
+        if (rainObject != null) {
+            weather.setRain(getRain(rainObject));
+            return;
+        }
+
+        if (snowObject != null) {
+            weather.setRain(getRain(snowObject));
+            return;
+        }
+
+        weather.setRain(0);
+    }
+
+    private static double getRain(@NonNull JSONObject rainObject) throws JSONException {
+        if (rainObject.has("3h")) {
+            return rainObject.getDouble("3h");
+        }
+
+        if (rainObject.has("1h")) {
+            return rainObject.getDouble("1h");
+        }
+
+        return 0;
+    }
+
+    private static void setSunsetAndSunrise(Weather weather, @NonNull JSONObject systemObject) throws JSONException {
+        if (systemObject.has("sunrise") && systemObject.has("sunset")) {
+            weather.setSunrise(systemObject.getString("sunrise"));
+            weather.setSunset(systemObject.getString("sunset"));
+        }
+    }
+
+    private static void setCityAndCountry(Weather weather, int cityId, @NonNull String cityName, @NonNull String country)
+    {
+        weather.setCityId(cityId);
+        weather.setCity(cityName);
+        weather.setCountry(country);
+    }
+
+    private static void setCoordinates(Weather weather, @NonNull JSONObject coordinatesObject) throws JSONException {
+        weather.setLat(coordinatesObject.getDouble("lat"));
+        weather.setLon(coordinatesObject.getDouble("lon"));
+    }
+
+    private static String capitalize(String string)
+    {
+        if (string.length() < 1) {
```

Comment:
`string.isEmpty()` has a clearer intention

Reply:
it does, isEmpty is better

Reply:
Fixed in 3629f50fbee9acd5ff180c741355bb80471d1f6b